guides/octane-auth-torii-with-github.md

```diff
@@ -0,0 +1,668 @@
+import Route from "@ember/routing/route";
+import AuthenticatedRouteMixin from "ember-simple-auth/mixins/authenticated-route-mixin";
+
+export default class IndexRoute extends Route.extend(AuthenticatedRouteMixin) {
+  model() {
+    return this.store.findRecord("github-user", "#");
+  }
+}
+```
+
+Here, the `#` is an id representing the currently logged in user for
+`ember-data-github`.
+
+
+```handlebars
+{{!-- app/templates/index.hbs --}}
+<h3>Index Route</h3>
+{{#if this.session.data.authenticated.authorizationCode}}
+  <p>Authorization Code: {{this.session.data.authenticated.authorizationCode}}</p>
+{{/if}}
+<img src={{this.model.avatarUrl}} alt="Avatar of {{this.model.login}}"> <br>
+Login: {{this.model.login}} <br>
+Full Name: {{this.model.name}}
+```
+
+If you launch the server now, you will get a blank page, and the console will
+tell you that the call to `https://api.github.com/user` was unauthorized. We
+can, however, extend the adapter for the `github-user` model and inject
+authorization.
+
+```sh
+ember generate adapter github-user
+```
+
+Now replace `app/adapters/github-user.js` with:
+
+```javascript
+// app/adapters/github-user.js
+import { computed } from "@ember/object";
+import GitHubUserAdapter from "ember-data-github/adapters/github-user";
+import DataAdapterMixin from "ember-simple-auth/mixins/data-adapter-mixin";
+
+export default GitHubUserAdapter.extend(DataAdapterMixin, {
+  headers: computed("session.data.authenticated.access_token", function() {
+    const headers = {};
+    if (this.session.isAuthenticated) {
+      headers.Authorization = `Bearer ${
+        this.session.data.authenticated.access_token
+      }`;
+    }
+
+    return headers;
+  })
+});
+```
+
+This adapter injects an authorization header into the GitHub request now.  The
+`DataAdapterMixin` is a mixin provided by `ember-simple-auth` that injects the
+`session` service. Otherwise, when you start up your server one last time, log
+in via the `login` route, and get dumped back to the `index` route, you should
+now see your avatar, your GitHub login name, and your full name.
+
+Congrats!
+
+## Next Steps
+
+This all done, you have a crude Ember Octane app that authenticates with
+GitHub, returning the logged in user. This model can then be used to populate
+your “real” `User` model, perhaps.
+
+Additionally, in production you cannot rely on using your local Gatekeeper
+server, but luckily Gatekeeper includes documentation on [deploying to
+Heroku](https://github.com/prose/gatekeeper#deploy-on-heroku) and similar
+services. Then, update the appropriate line in your Ember app’s `.env` to
+point to the Heroku url, not `http://localhost:9999/authenticate`.
+
+Finally, I’ve created a repo of the “finished” version of this guide, and it
+is available at
+[@muziejus/ember-simple-auth-github-octane](http://github.com/muziejus/ember-simple-auth-github-octane).
+
+Here are some other links provided by [@srvance](http://github.com/srvance),
```

Comment:
It's really cool that you added further resources for learning and the related credit here :+1: